File: user-service/src/main/java/com/transport/urbain/userservice/event/UserRegisteredEvent.java
package com.transport.urbain.userservice.event;

import java.io.Serializable;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor; 

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserRegisteredEvent implements Serializable {
    private String userId;
    private String email;
    private String firstName;
    private String lastName;
}

File: notification-service/src/notifications/dto/user-registered.event.ts
import { IsString, IsEmail, IsNotEmpty } from 'class-validator';

export class UserRegisteredEventDto {
    @IsString()
    @IsNotEmpty()
    userId: string;

    @IsEmail()
    @IsNotEmpty()
    email: string;

    @IsString()
    @IsNotEmpty()
    firstName: string;

    @IsString()
    @IsNotEmpty()
    lastName: string;
}